### DISCUSSION TITLE
Question about TypeConverterException vs. ReaderException

### DISCUSSION THREAD

**Discussion by ILTCED** [JoshClose/CsvHelper#2051] [Q&A]
I have catch block that uses CsvHelperException to catch everything possible. I then check to see which type of exception the error really is and handle it accordingly. 

I notice when I put invalid data into a datetime formatted model variable, a ReaderException occurs. However, if I put invalid data into a decimal formatted model variable, a TypeConversionException occurs.

1. Why is that?
2. Is there any documentation that specifies the scenarios in which a certain exception will occur?

**Comment by JoshClose** [comment]
What version are you using? That was a bug that was fixed in version 29.0.0 released yesterday to NuGet.
